File: src/main/java/corp/soyummy/userfavorites/dto/UserFavoriteCreateDto.java
package corp.soyummy.userfavorites.dto;

import lombok.Getter;
import lombok.Setter;
import lombok.AllArgsConstructor;

import jakarta.validation.constraints.NotNull;
import jakarta.validation.constraints.Size;

@Getter
@Setter
@AllArgsConstructor
public class UserFavoriteCreateDto {
    @NotNull(message = "Recipe ID is required!")
    @Size(min = 1, message = "RecipeId must have at least one character.")
    private String recipeId;

    @NotNull(message = "User ID is required!")
    @Size(min = 1, message = "UserId must have at least one character.")
    private String userId;
}

File: src/main/kotlin/com/soyummy/userfavorites/dto/UserFavoriteCreateDto.kt
package com.soyummy.userfavorites.dto

import jakarta.validation.constraints.NotBlank
import jakarta.validation.constraints.Size

data class UserFavoriteCreateDto(
    @field:NotBlank(message = "Recipe ID is required!")
    @field:Size(min = 1, message = "RecipeId must have at least one character.")
    val recipeId: String,

    @field:NotBlank(message = "User ID is required!")
    @field:Size(min = 1, message = "UserId must have at least one character.")
    var userId: String
)